src/app/home-module/services/student-service.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { Student } from '../models/student';
+import { environment } from 'src/environments/environments';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StudentServiceService {
+  BaseUrl = environment.BaseUrL;
+
+  constructor(private http: HttpClient) {}
+
+  public ageCalculation(birthday: Date): number {
+    const today = new Date();
+    const birthDate = new Date(birthday);
+    let age = today.getFullYear() - birthDate.getFullYear();
+    const m = today.getMonth() - birthDate.getMonth();
+    if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
+      age--;
+    }
+
+    return age;
+  }
+
+  public getStudents(): Observable<Student[]> {
```

Comment:
dont have to subscribe .you can use like this .`return this.http.get<Student[]>(this.BaseUrl)`